From d9d9188541bdd8e498e11f4c0dd3a93f32ac72d7 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 1 Aug 2024 17:36:44 +0300
Subject: [PATCH 02/18] arm64: renesas: solidrun: move scif2 pinmux to
 som-specific dtsi

scif2 pinctrl is different between some Renesas chips.
Remove it from shared dtsi and add explicitly to g2l and v2l som dtsi.
g2lc was already overriding the shared pinctrl, leading to confusion.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm64/boot/dts/renesas/rz_2l-sr-pinfunction.dtsi |  7 -------
 arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi         |  9 +++++++++
 arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi        | 10 +++++-----
 arch/arm64/boot/dts/renesas/rzv2l-sr-som.dtsi         |  8 ++++++++
 4 files changed, 22 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/rz_2l-sr-pinfunction.dtsi b/arch/arm64/boot/dts/renesas/rz_2l-sr-pinfunction.dtsi
index 79cacfebd..d335190a6 100644
--- a/arch/arm64/boot/dts/renesas/rz_2l-sr-pinfunction.dtsi
+++ b/arch/arm64/boot/dts/renesas/rz_2l-sr-pinfunction.dtsi
@@ -93,13 +93,6 @@ scif1_pins: scif1 {
 			 <RZG2L_PORT_PINMUX(41, 1, 1)>; /* RTS# */
 	};
 
-	scif2_pins: scif2 {
-		pinmux = <RZG2L_PORT_PINMUX(48, 0, 1)>, /* TxD */
-			 <RZG2L_PORT_PINMUX(48, 1, 1)>, /* RxD */
-			 <RZG2L_PORT_PINMUX(48, 3, 1)>, /* CTS# */
-			 <RZG2L_PORT_PINMUX(48, 4, 1)>; /* RTS# */
-	};
-
 #if SW_SD0_DEV_SEL
 	sdhi0_emmc_pins: sd0emmc {
 		sd0_emmc_data {
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
index 9e2c412c7..a53c7c015 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
@@ -71,3 +71,12 @@ MXL8611X_LEDX_CFG_LINK_UP_10MB_ON
                 mxl-8611x,tx-internal-delay-ps-1g = <300>;
         };
 };
+
+&pinctrl {
+	scif2_pins: scif-2-pins {
+		pinmux = <RZG2L_PORT_PINMUX(48, 0, 1)>, /* SCIF2_TXD */
+			 <RZG2L_PORT_PINMUX(48, 1, 1)>, /* SCIF2_RXD */
+			 <RZG2L_PORT_PINMUX(48, 3, 1)>, /* SCIF2_CTS# */
+			 <RZG2L_PORT_PINMUX(48, 4, 1)>; /* SCIF2_RTS# */
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
index 15251ae1f..f68daa99c 100644
--- a/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
@@ -122,11 +122,11 @@ i2c2_pins: i2c2 {
                          <RZG2L_PORT_PINMUX(42, 4, 1)>; /* SCL */
         };
 
-	scif2_pins: scif2 {
-		pinmux = <RZG2L_PORT_PINMUX(42, 0, 4)>, /* TxD */
-			 <RZG2L_PORT_PINMUX(42, 1, 4)>, /* RxD */
-			 <RZG2L_PORT_PINMUX(5, 1, 2)>, /* CTS# */
-			 <RZG2L_PORT_PINMUX(5, 2, 2)>; /* RTS# */
+	scif2_pins: scif-2-pins {
+		pinmux = <RZG2L_PORT_PINMUX(42, 0, 4)>, /* SCIF2_TXD */
+			 <RZG2L_PORT_PINMUX(42, 1, 4)>, /* SCIF2_RXD */
+			 <RZG2L_PORT_PINMUX(5, 1, 2)>, /* SCIF2_CTS# */
+			 <RZG2L_PORT_PINMUX(5, 2, 2)>; /* SCIF2_RTS# */
 	};
 };
 
diff --git a/arch/arm64/boot/dts/renesas/rzv2l-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzv2l-sr-som.dtsi
index 8886c48f1..c97cf7237 100644
--- a/arch/arm64/boot/dts/renesas/rzv2l-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzv2l-sr-som.dtsi
@@ -106,3 +106,11 @@ &drpai0 {
 	status = "okay"; 
 };
 
+&pinctrl {
+	scif2_pins: scif-2-pins {
+		pinmux = <RZG2L_PORT_PINMUX(48, 0, 1)>, /* SCIF2_TXD */
+			 <RZG2L_PORT_PINMUX(48, 1, 1)>, /* SCIF2_RXD */
+			 <RZG2L_PORT_PINMUX(48, 3, 1)>, /* SCIF2_CTS# */
+			 <RZG2L_PORT_PINMUX(48, 4, 1)>; /* SCIF2_RTS# */
+	};
+};
-- 
2.43.0

